.github/workflows/ci.yml

```diff
@@ -0,0 +1,171 @@
+name: CI 
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, closed]
+    branches:
+      - master
+env:
+  AWS_REGION: "eu-central-1"
+  SSH_AUTH_SOCK: /tmp/ssh_agent.sock
+
+permissions:
+  id-token: write
+  contents: write
+
+jobs:
+  earthly-build-and-push:
+    runs-on: ubuntu-latest
+    if: github.event.action != 'closed' || github.event.pull_request.merged == true
+    permissions:
+      id-token: write
+      contents: write
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup Earthly
+        uses: ./.github/earthly-setup
+        with:
+          ssh_key: ${{ secrets.SUBSTRATE_REPO_SSH_KEY }}
+          config_tar: ${{ secrets.EARTHLY_TAR }}
+      - name: Acquire AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN_SECRET }}
+          aws-region: ${{ env.AWS_REGION }}
+      - name: Login to container registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ secrets.ECR_REGISTRY_SECRET }}
+      - name: Build and Push to ECR
+        env:
+          EARTHLY_CI: true
+          EARTHLY_OUTPUT: true 
+          EARTHLY_PUSH: true
+        run: |
+          earthly -P +ci --image=${{ secrets.ECR_REGISTRY_SECRET }}/substrate-node
+      - name: Upload chain spec artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: chain-specs
+          path: |
+            ./devnet_chain_spec.json
+            ./staging_preview_chain_spec.json
+            ./staging_preprod_chain_spec.json
+
+  build-artifacts:
```

Comment:
can't we just upload the build artifacts from the previous +build step?

Reply:
Yes I need to rewrite so we have a single earthly invocation that builds and exports:
- ECR docker image
- partner-chains-node
- partner-chains-cli
- chain specs for all 3 environments

Reply:
Done I think!